src/index.ts

```diff
@@ -44,25 +60,124 @@ export class WalletPluginMetaMask extends AbstractWalletPlugin implements Wallet
         })
     }
 
+    async lookupAccounts(publicKey: PublicKey, chainId: Checksum256Type): Promise<AccountFound[]> {
+        const response = await fetch(
+            `${ACCOUNT_LOOKUP_URL}/lookup/${publicKey.toLegacyString()}?includeTestnets=true`
+        )
+        const accountsByNetwork: AccountLookup[] = await response.json()
+
+        const networkAccount = accountsByNetwork.find(
+            (networkAccount: AccountLookup) => networkAccount.chainId === String(chainId)
+        )
+
+        if (!networkAccount) {
+            return []
+        }
+
+        return networkAccount.accounts
+    }
+
+    async getPermissionLevel(
+        context: LoginContext,
+        chain: ChainDefinition
+    ): Promise<PermissionLevel> {
+        if (context.permissionLevel) {
+            return context.permissionLevel
+        }
+
+        const activeKey = await this.retrievePublicKey(chain.id, 1)
```

Comment:
I don't think there's any value in deriving the owner and active key from different parts of the mnemonic. 

Both of those keys are derived from the same place, and stored in the same place... so if one gets stolen the other will too. 

For these accounts I think we should just use the same key for both. Maybe in the future we could add a feature to generate an owner key certificate or something, so that they'd actually have two separate keys. 

So here it'd just:

- Make the keys both default to index 0.
- Remove the index parameter from `retrievePublicKey` (since its unused)

Reply:
Hmm you may be right.. it's good practice to use different keys but in this case since the keys can't be exported I guess there is no risk of the active key being leaked somehow 🤔